### PR TITLE
add sample weights to scikit-learn interface.

### LINKED ISSUE #17: Feature request: Sample weights
In [this work](https://gitlab.com/miguel.romao/symbolic-regression-bsm) ([preprint](https://arxiv.org/abs/2405.18471)), we had to resample the dataset so that the target variable had a flat distribution to prevent `pyoperon` from only learning about the most common values. A better alternative, that `pyoperon` does not currently support, is to use `sample_weights` when computing the loss function.

For example, we would have `X`, `y`, `w` (sample weights), then the loss function is computed as the weighted average of the losses of each data instance
$$loss_{total} = \frac{1}{\sum_i w_i} \sum_i w_i \times loss(X_i, y_i)$$
this is already supported by many `scikit-learn` estimators.

The use case would be varied: from specifying the focus of a regression task, to use datasets that have been sampled in a way that the data come with sampling weights (this is common practice in social sciences and in simulation-based inference in STEM).

### PR DESCRIPTION
**Pull request by LukasCamera**
Adds sample_weight for error metrics like r2, mse, etc. to the scikit-learn wrapper. Sample weights passed to the `fit` method to be compatible with scikit-learn pipelines.

Closes #17. Requires pull request https://github.com/heal-research/operon/pull/49. #

### CODE REVIEW

**Comment by foolnotion** [comment]
Superseded by #65, which implements the same goal (`sample_weight` support for `SymbolicRegressor.fit()`, closing the same #17) with full end-to-end coverage (including the coefficient-optimizer weighting fix on the operon side, heal-research/operon#104, now merged) and green CI across all platforms/Python versions. Closing this one in favor of #65.
